clang/lib/Basic/CMakeLists.txt

```diff
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  Core
```

Comment:
This is needed because `TargetInfo.cpp` now needs LLVM types.

Reply:
Does this create additional dependencies on clang based tools that don't need to use LLVM IR? For example, clang-format?

Reply:
This shouldn't be necessary. I think this code is in the wrong place, I would expect it to go in lib/CodeGen/Targets, not lib/Basic

Reply:
> This shouldn't be necessary. I think this code is in the wrong place, I would expect it to go in lib/CodeGen/Targets, not lib/Basic

And then to have clang depend on LLVM's CodeGen component? If that's the case, I don't think that is correct. That is even worse.

> Does this create additional dependencies on clang based tools that don't need to use LLVM IR? For example, clang-format?

That's a good question. I think it does. Do you have suggestions on how to design the interface?

Reply:
No, this is clang lib/CodeGen which has nothing to do with actual codegen. Someone really needs to just go ahead with the rename to IRGen

Reply:
> No, this is clang lib/CodeGen which has nothing to do with actual codegen. Someone really needs to just go ahead with the rename to IRGen

Core is LLVM's lib/IR isn't it? This is making clang lib/Basic depend on llvm/lib/IR. It can't be making Basic depend on clang's lib/CodeGen, that would be a circular dependency.

Reply:
This should be defined here https://github.com/llvm/llvm-project/blob/main/clang/lib/CodeGen/Targets/AMDGPU.cpp